### DISCUSSION TITLE
Repo Language not detected, Markdown.

### DISCUSSION THREAD

**Discussion by Cats1337** [github-linguist/linguist#6032] [Classification] [answered]
## No Language Showing?
My Repo: https://github.com/Cats1337/TF2Bots
There is no "language statistic bar" though it's written almost entirely in markdown.
I've been looking at linguist and am not sure why it doesn't show.

Cheers for whoever helps.

_Edit: Oops! Wasn't supposed to be put in improvement, and I can't change the label, as far as I'm aware..._

**Comment by lildude** [comment, marked as answer]
> There is no "language statistic bar" though it's written almost entirely in markdown.

You've just answered your own question. 

From [How linguist works](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md):

> Linguist starts by going through all the files in a repository and excludes all files that it determines to be binary data, [vendored code](https://github.com/github/linguist/blob/master/docs/overrides.md#vendored-code), [generated code](https://github.com/github/linguist/blob/master/docs/overrides.md#generated-code), [documentation](https://github.com/github/linguist/blob/master/docs/overrides.md#documentation), or are defined as `data` (e.g. SQL) or `prose` (e.g. Markdown) languages, whilst taking into account any [overrides](https://github.com/github/linguist/blob/master/docs/overrides.md).

As stated, Markdown is `prose` so excluded by default.

If you want it to be counted, you'll need to implement an [override](https://github.com/github/linguist/blob/master/docs/overrides.md).

As an aside, I've converted this to a discussion as this is where you really should have started... it even says so on the list of issue templates you we present with when opening the original issue 😉

**Reply by Cats1337** [reply]
I see, is there a way to enable it to display markdown?
The other I have is https://github.com/Cats1337/ChatFilter which for some reason isn't displaying that it's Java, is it because of the file structure or something different?

Also, my bad for wrong section, I've never really used the issue / discussion area of GitHub so I'm new to this area.

**Reply by lildude** [reply]
> I see, is there a way to enable it to display markdown?

Yup. Using an override as I mentioned and linked to before.

> The other I have is https://github.com/Cats1337/ChatFilter which for some reason isn't displaying that it's Java, is it because of the file structure or something different?

No. This is because all of your Java files are binaries, eg:

```console
$ file module-info.class
module-info.class: compiled Java class data, version 60.0
$
```

Language detection only happens for source code files, not compiled binaries.
